bundle/edu.gemini.p2checker/src/main/scala/edu/gemini/p2checker/rules/ghost/GhostRule.scala

```diff
@@ -0,0 +1,46 @@
+package edu.gemini.p2checker.rules.ghost
+
+import edu.gemini.p2checker.api.{IP2Problems, IRule, ObservationElements, P2Problems}
+import edu.gemini.shared.util.immutable.ScalaConverters._
+import edu.gemini.spModel.core.Coordinates
+import edu.gemini.spModel.gemini.ghost.GhostScienceAreaGeometry
+
+object GhostRule extends IRule {
+  object CoordinatesOutOfFOVRule extends IRule {
+    override def check(elems: ObservationElements): IP2Problems = {
+      val problems = new P2Problems()
+
+      // COORDINATES, i.e. sky positions
+      for {
+        toc <- elems.getTargetObsComponentNode.asScalaOpt
+        ctx <- elems.getObsContext.asScalaOpt
+        env <- Option(ctx.getTargets)
+        base <- env.getAsterism.basePosition(None)
+        c <- env.getCoordinates.asScalaList
+        if Coordinates.difference(base, c.coordinates).distance.toArcsecs > GhostScienceAreaGeometry.radius.toArcsecs
+      } problems.addError(GhostRule.Prefix + "CoordinatesOutOfRange", String.format(GhostRule.CoordinatesOutOfRange, c.getName), toc)
+
+      // TARGETS
+      for {
+        toc <- elems.getTargetObsComponentNode.asScalaOpt
+        ctx <- elems.getObsContext.asScalaOpt
+        env <- Option(ctx.getTargets)
+        base <- env.getAsterism.basePosition(None) //ctx.getBaseCoordinates.asScalaOpt
+        t <- env.getAsterism.allTargets
+        c <- t.coords(None)
+        if Coordinates.difference(base, c).distance.toArcsecs > GhostScienceAreaGeometry.radius.toArcsecs
+      } problems.addError(GhostRule.Prefix + "CoordinatesOutOfRange", String.format(GhostRule.CoordinatesOutOfRange, t.name), toc)
+
+      problems
+    }
+  }
+
+  override def check(elems: ObservationElements): IP2Problems = {
+    val probs: P2Problems = new P2Problems()
+    probs.append(CoordinatesOutOfFOVRule.check(elems))
+    probs
+  }
+
+  val Prefix: String = "GhostRule_"
+  val CoordinatesOutOfRange: String = "The coordinates for %s are out of range of the base position."
```

Comment:
Maybe "out of range *at* the base position"?  Or was this prescribed in a requirement somewhere?  It's the most trivial of quibbles but the base position isn't reaching for anything.

Reply:
It definitely wasn't stated specifically, and I think your wording is better, so we'll go with that.